pkg/k8shandler/index_management.go

```diff
@@ -0,0 +1,128 @@
+package k8shandler
+
+import (
+	"fmt"
+	"strings"
+
+	logging "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/indexmanagement"
+	"github.com/openshift/elasticsearch-operator/pkg/logger"
+	esapi "github.com/openshift/elasticsearch-operator/pkg/types/elasticsearch"
+)
+
+const (
+	//ocpTemplatePrefix is the prefix all operator generated templates
+	ocpTemplatePrefix = "ocp-gen"
+)
+
+func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateIndexManagement() error {
+
+	logger.Debug("Reconciling IndexManagement")
+	cluster := elasticsearchRequest.cluster
+	if cluster.Spec.IndexManagement == nil {
+		logger.Debug("IndexManagement not specified - noop")
+		return nil
+	}
+	indexManagement := indexmanagement.VerifyAndNormalize(cluster)
+	//TODO find crons with no matching mapping and remove them
+	elasticsearchRequest.cullIndexManagement(indexManagement.Mappings)
+	for _, mapping := range indexManagement.Mappings {
+		logger.Debugf("reconciling index management for mapping: %s", mapping.Name)
+		//create or update template
+		if err := elasticsearchRequest.createOrUpdateIndexTemplate(mapping); err != nil {
+			logger.Errorf("Error creating index template for mapping %s: %v", mapping.Name, err)
+			return err
+		}
+		//TODO: Can we have partial success?
+		if err := elasticsearchRequest.initializeIndexIfNeeded(mapping); err != nil {
+			logger.Errorf("Error intializing index for mapping %s: %v", mapping.Name, err)
+			return err
+		}
+	}
+	// for _, policy := range indexManagement.Policies {
+	// 	//deploy rollover cron
+
+	// 	//deploy delete cron
+	// }
+
+	return nil
+}
+func (elasticsearchRequest *ElasticsearchRequest) cullIndexManagement(mappings []logging.IndexManagementPolicyMappingSpec) {
+	mappingNames := map[string]interface{}{}
+	for _, mapping := range mappings {
+		mappingNames[formatTemplateName(mapping.Name)] = nil
+	}
+	existing, err := elasticsearchRequest.ListTemplates()
+	if err != nil {
+		logger.Warnf("Unable to list existing templates in order to reconcile stale ones: %v", err)
+		return
+	}
+	for _, template := range existing {
+		found := false
+		for name := range mappingNames {
+			if name == template {
+				found = true
+				break
+			}
+		}
```

Comment:
could we write/use a function that finds the diff of these lists instead and returns them? it seems like there would be a quick way to do that in golang already...

Reply:
updated to user k8s set

Reply:
use kube sets